ifrs17-template/Test/IfrsVariablesTest.ipynb

```diff
@@ -183,21 +264,29 @@
             "source": [
                 "public List<BenchmarkTestResult> CompareAgainstBm(List<BenchmarkTestResult> errors, IEnumerable<IfrsVariable> expected, IEnumerable<IfrsVariable> computed)",
                 "\n{",
-                "\n    var expectedNotComputed = expected.Where(x => Math.Abs(x.Value) > BenchmarkPrecision).Except(computed, EqualityComparer<IfrsVariable>.Instance);",
-                "\n    var computedNotExpected = computed.Where(x => Math.Abs(x.Value) > BenchmarkPrecision).Except(expected, EqualityComparer<IfrsVariable>.Instance);",
+                "\n    var comparer = EqualityComparer<IfrsVariable>.Instance;",
+                "\n    var expectedNotComputed = expected.Where(x => Math.Abs(x.Value) > BenchmarkPrecision).Except(computed, comparer);",
+                "\n    var computedNotExpected = computed.Where(x => Math.Abs(x.Value) > BenchmarkPrecision).Except(expected, comparer);",
                 "\n    if (expectedNotComputed.Any())",
                 "\n        foreach (var variable in expectedNotComputed)",
                 "\n            errors.Add(new BenchmarkTestResult(\"Extra expected variable for: Partition \" + variable.Partition + \", \" + variable.ToIdentityString()));",
                 "\n    if (computedNotExpected.Any())",
                 "\n        foreach (var variable in computedNotExpected)",
                 "\n            errors.Add(new BenchmarkTestResult(\"Extra computed variable for: Partition \" + variable.Partition + \", \" + variable.ToIdentityString()));",
                 "\n",
-                "\n    var comparer = EqualityComparer<IfrsVariable>.Instance; ",
-                "\n    foreach (var benchmarkVariable in expected) {",
-                "\n        var computedVariable = computed.FirstOrDefault(y => comparer.Equals(benchmarkVariable,y))?? null;",
-                "\n        if(computedVariable == null || Math.Abs(benchmarkVariable.Value + computedVariable.Value) < BenchmarkPrecision) continue;",
-                "\n        errors.Add(new BenchmarkTestResult(\"Value does not match for Variable: Partition \" + benchmarkVariable.Partition + \", \" + benchmarkVariable.ToIdentityString(), -benchmarkVariable.Value, computedVariable.Value));",
-                "\n    }",
+                "\n    var benchmarkErrors = expected.Concat(computed)",
+                "\n            .GroupBy(x => new{Partition = x.Partition, IdentityString = x.ToIdentityString(listIfrsValueProperties)})",
```

Comment:
Do you need this input? ValueN prop are not IdentityProperty. 
Also consider switching to use the method ToStringWith (Extentions NB) which does not rely on reflection

Reply:
Done.